### ISSUE TITLE
Test audio↔Julia morphology alignment with independent geometry, color, and glow

### ISSUE THREAD

**Issue by rhythmatician** [rhythmatician/FractalSync#140]
Part of #77.

Coordinates with #105 (roughness owner), #95 (color/glow presentation), #108/#109 (observation/audio contracts), #103 (offline teachers), #111 (evaluation cockpit), #138 (navigation), #139 (Julia Probe Instrument), and #128/#132 (evidence-gated reward/training).

## Goal

Test whether a small, interpretable morphology feature space helps the Player choose Julia regions that fit the music's timbre and texture, independently of palette and edge glow.

Working decomposition:

```text
musical temporal structure -> motion / cycles of the Julia probe
musical timbre / texture   -> morphology of the Julia image
color                     -> independent restrained presentation state
glow                      -> independent one-dimensional presentation state
```

These are separable controls and experiment factors, not claims of perceptual independence or hard-coded audio-to-render rules. The Player may coordinate them. Geometry is the intended primary structural carrier, color a complementary carrier, and glow a small additional expressive dimension. Existing Julia zoom/rotation controls remain separate; this decomposition does not remove them.

The user explicitly wants to retain the renderer's edge-glow capability. Exact glow semantics belong in #95 and the versioned presentation/control contract, not in the morphology metric.

## Scope / ownership

This issue owns the cross-axis feature matrix, common diagnostic representation, controlled comparison protocol, and keep/revise/drop verdicts.

**#105 remains the detailed roughness research/implementation owner.** Reuse its audit, candidates, tests, and eventual extractor rather than implementing a second roughness metric here.

Color/brightness and glow/diffusion are companion hypotheses, coordinated with #95. They must not confound the geometry experiment. #109 owns any eventual causal audio features; #108 decides which evidence earns observation channels. A useful evaluator is not automatically a runtime input or reward.

## Candidate feature matrix

All entries below are hypotheses and candidate measurements, not established cross-modal laws.

| Perceptual axis | Audio-side candidates | Julia geometry candidates | Presentation controls / confounds | Initial test |
|---|---|---|---|---|
| Rough ↔ smooth | psychoacoustic roughness; inharmonicity; flatness/noisy energy, measured separately | fine-scale wavelet energy; multiscale boundary irregularity; edge density / gradient energy | palette contrast and glow can change apparent roughness | H1: rougher audio is judged to fit rougher morphology; consume #105 |
| Fat ↔ thin | low/mid spectral mass; harmonic fullness; bandwidth and partial occupancy | robust local/filament width; distance-to-boundary along a defined skeleton; body thickness | glow, framing, and zoom can inflate apparent thickness | H2: fuller audio is judged to fit thicker structures independently of density |
| Orderly/stable ↔ irregular/agitated | harmonicity and tonal stability; spectral/wavelet energy entropy; temporal irregularity kept separate | symmetry and repetition scores; branching regularity; multiscale entropy; lacunarity as a separate gap-distribution descriptor | color instability and temporal jitter must not substitute for structural disorder | H3: musical order/irregularity has a useful morphology counterpart; test entropy and harmonic/structural-order subhypotheses separately |
| Dense ↔ sparse | active-band occupancy; simultaneous spectral activity; onset/event density as a separate temporal candidate | occupied-area fraction; branch/feature count per area; boundary length per area | crop, glow, and contrast can inflate occupancy or feature count | H4: denser audio is judged to fit denser morphology independently of thickness and roughness |
| Bright ↔ dark | spectral centroid; high-frequency energy ratio; rolloff | fine/coarse spatial-frequency balance only as a separate ablation | lightness is the primary presentation candidate; chroma and glow tested independently | companion hypothesis, not a compulsory mapping from audio brightness to c |
| Dry/crisp ↔ diffuse/blooming | attack-to-tail behavior; sustain/decay; candidate reverberant/diffuse or high-frequency-air evidence | no required morphology change | independent scalar edge glow; distinguish it from overall lightness | companion hypothesis: glow may communicate sustain/diffusion beyond brightness |

Keep raw candidate descriptors separate initially. Do not prematurely construct weighted scalar scores such as `order = symmetry - entropy` or declare `m_audio = m_Julia` the final objective.

Fatness and density need explicit counterexamples: a few thick lobes versus many thin branches; a full-bodied solo sound versus a busy thin texture. Likewise, roughness, noisiness, brightness, and loudness are not interchangeable labels.

## Canonical geometry measurement

Do not extract the canonical morphology score from the final RGB framebuffer.

Use a reproducible pre-palette/pre-glow Julia representation, such as a specified escape/distance scalar field or canonical monochrome mask. Record exactly what is being measured: a finite-resolution boundary neighborhood, a filled-set approximation, an escape-field band, or another stated construction. Do not ambiguously call all of those 'J'.

For width/skeleton/occupancy measurements, specify foreground definition, band thickness or threshold, normalization, and empty/degenerate-mask behavior. Render-resolution or threshold artifacts must not masquerade as physical filament thickness.

Record known `c_J`, Julia viewport center/zoom/rotation, resolution, iteration budget, escape convention, antialiasing, and extractor version. Start with a fixed view protocol, then separately test sensitivity to view changes and multiple resolutions. Do not claim a descriptor is intrinsically view-invariant without evidence.

Candidate measurements include:

- normalized wavelet energy by spatial scale; fine/total energy ratio;
- boundary length, gradient energy, and local irregularity at specified scales;
- robust width statistics from a precisely defined support mask;
- occupied fraction, branch density, and feature count;
- symmetry/repetition and branching consistency;
- lacunarity across scales, measuring gap distribution rather than treating it as another name for entropy;
- macro-form versus micro-detail contrast.

Check whether proposed symmetry features actually distinguish the sampled Julia family rather than rewarding a symmetry shared by all examples. Do not infer exact connectedness, symmetry order, or parameter values from colored screenshots alone.

## Entropy: test correspondence, not numerical coincidence

A candidate baseline uses normalized audio spectral/wavelet energy `p_k` and image spatial-frequency/wavelet energy `q_j`:

```text
H_A = -sum p_k log(p_k) / log(N_A)
H_J = -sum q_j log(q_j) / log(N_J)
```

This puts both statistics in [0,1]; it does **not** establish semantic equivalence. Define bins, scale weights, windowing, bandwidth, foreground/background handling, and low-energy validity. Do not assign 'orderly' to silence merely because a normalization failed.

`abs(H_A-H_J)` is a null/simple candidate to test, not a pre-approved loss. Compare rank/trend agreement and a small calibration fitted on training data. Include equal-entropy examples with different perceived organization and power-spectrum-preserving phase/randomization controls where applicable. Spectral-energy entropy alone must not be assumed to measure spatial arrangement, consonance, temporal unpredictability, or perceptual complexity.

### 1. Audit and reproducible examples

Reuse #105's legacy/extractor audit. Inspect the current Julia renderer and identify the pre-palette field and existing edge-glow control. Build a small diverse set of known-c renders and controlled audio excerpts. The images discussed with the user motivate hypotheses but are not numerical ground truth for c, topology, or descriptor values.

Start with diagnostic stimuli and representative music, including cases that separate fullness, density, brightness, roughness, and noise. Keep audio level and visual framing controlled where they would reveal the answer trivially.

### 2. Validate descriptors within each modality

Ask whether candidate measurements agree with human judgments of roughness, width, density, and organization before testing cross-modal fit. Measure redundancy and sensitivity; retain small interpretable feature sets.

### 3. Test cross-modal fit before training a full policy

Use blinded/randomized pairwise audio-image or audio-short-loop comparisons. Compare predicted matches, deliberate opposite-axis matches, and shuffled matches. Permit ties/no meaningful difference.

Hold palette and glow fixed for the four morphology hypotheses. For color tests hold geometry/glow fixed; for glow tests hold geometry/palette fixed, and include brightness-matched comparisons to test whether glow adds more than lightness.

Split by song/performance and parameter neighborhoods, not adjacent frames of the same song/trajectory. Fit normalization/calibration on training splits only. Report disagreement and uncertainty, not just a mean preference score.

### 4. Test motion and probe composition

After static/short-loop evidence, evaluate coherent trajectories without sacrificing existing temporal smoothness. With #139 active, score the actual rendered parameter `c_J(t)`, not only rider position `b(t)`.

Compare a matched neighborhood throughout a probe cycle against a path that has a good average score but visits inappropriate morphology repeatedly. Keep rhythmic entrainment evidence separate from timbral/morphological fit. No hard 30-Hz ontology is required: audible carrier spectrum and temporal modulation are different descriptor roles.

### 5. Feed supported results into learning

Only supported axes become candidate objectives or region-selection evidence. Compare no-alignment baseline, one axis at a time, and the smallest useful combination. Log terms separately from #128 temporal entrainment, continuity, topology, control costs, palette, and glow.

An appearance atlas is a possible follow-up, not a prerequisite. Because the Julia parameter has only two real degrees of freedom at a fixed view, do not assume four morphology coordinates are independently controllable or that every desired feature vector is reachable. Measure the attainable combinations and local sensitivity. #138 handles travel to a supplied region; it must not turn a descriptor target into a direct-c servo.

## Architecture and cost

DSP measures causal evidence; PlayerPolicy chooses regions, motions, palette, and glow through explicit contracts. No fixed descriptor-to-c lookup or parallel brightness/glow policy is introduced.

Rust `runtime-core` remains authoritative for deterministic semantics shared by runtime/training/evaluation. Exploratory analysis and offline teachers stay explicitly research-only; any training surrogate must follow ADR 0001/#119. Version feature definitions and provenance.

Begin with offline or low-resolution diagnostic extraction. Profile before adding morphology passes to the live cockpit. DANTE's UHD 770 should not inherit a full-resolution per-frame wavelet/skeleton pipeline merely because it is useful for research. Keep cached/static-atlas approximations distinguishable from measurements of the live view.

## Deliverables / acceptance

- [ ] A reproducible feature matrix and protocol, with definitions, units, temporal/spatial support, normalization, validity, and version metadata.
- [ ] A canonical pre-color/pre-glow representation and explicit width/occupancy semantics.
- [ ] Palette/glow interventions leave canonical geometry measurements unchanged; zoom/resolution/threshold sensitivity is reported honestly.
- [ ] H1 is coordinated with #105 without a duplicate roughness authority.
- [ ] H2 fullness/thickness is tested independently of density and brightness.
- [ ] H3 order/irregularity tests entropy and harmonic/structural correspondence separately, including counterexamples.
- [ ] H4 density is tested independently of width, noise, and event rate.
- [ ] Brightness/color and glow/diffusion receive separate companion verdicts with cross-channel controls.
- [ ] Held-out blinded fit judgments beat appropriate shuffled/mismatched baselines for every retained claim, or a revise/drop verdict is recorded.
- [ ] Reachable morphology combinations and probe-cycle behavior are characterized before assuming an arbitrary target vector is realizable.
- [ ] Any retained loss is ablated against no alignment and checked for jitter, maximum-detail collapse, dust/occupancy hacks, and presentation shortcuts.
- [ ] #111 can replay the exact evidence with raw/normalized features, independent channel controls, and honest STATE/DIAG/TEACHER/OBS provenance.
- [ ] #108/#109 receive evidence-backed proposals only; no speculative tensor expansion is treated as complete.

## Non-goals

No universal equation equating timbre to fractal geometry; no unconditional entropy-equality loss; no fixed music-to-c/color/glow mapping; no relabeling Julia zoom as Mandelbrot scale; no new Physics law; no requirement to finish navigation or full RL before testing the perceptual hypotheses.

**Comment by rhythmatician** [comment]
2026-09-07 consonance/chord-quality extension: add a harmonic-organization subhypothesis to the orderly/stable ↔ irregular/agitated axis, but keep it distinct from sensory roughness and raw entropy.

A promising implementation reuses the complex Morlet/log-frequency representation. Let

```text
x = log2(f)
P(t,x) = smoothed/compressed spectral salience from |W(t,f)|
```

and define a pairwise ratio field

```text
A2(delta,t) = ∫ P(t,x) P(t,x+delta) dx.
```

Because ratios become translations in log-frequency, this is transposition-invariant and can measure concentration around simple frequency ratios without identifying notes or chords. A low-complexity small-integer ratio score is a candidate continuous harmonicity/consonance descriptor.

However, pairwise ratios are not enough for chord quality. Major-like and minor-like triads can share the same pairwise interval multiset. Therefore also test ordered third-order relation structure

```text
A3(d1,d2,t) = ∫ P(t,x) P(t,x+d1) P(t,x+d1+d2) dx
```

with orientation-sensitive summaries, e.g. the difference between `(log2(5/4), log2(6/5))` and the swapped ordering, as a continuous harmonic-chirality/polarity candidate rather than a major/minor classifier.

For #140, the visual questions become at least two separate hypotheses:

```text
harmonicity / simple-ratio organization
    ↔ Julia structural order / symmetry / regularity

harmonic orientation / chirality
    ↔ some continuous appearance polarity, possibly morphology and/or presentation
```

Do not assume the second belongs to lightness just because major often sounds 'brighter' than minor; test geometry, color, and glow independently under the existing controlled-intervention protocol.

Keep sensory roughness under #105 as a different axis: distorted but harmonically organized audio may have high roughness and high harmonicity simultaneously. Likewise, normalized audio/image entropy is not a substitute for harmonic organization.

This is another reason to prefer a small relational feature family over named chord classes. It also potentially connects to #139: temporal polyrhythm ratios and audible harmonic ratios are the same mathematical object in `log f`, so a later experiment could ask whether supported audio ratio relations are expressed naturally through nested probe-oscillator ratios without hard-coded chord semantics.

**Comment by rhythmatician** [comment]
Coordinate this ticket with new **#150**. #140 still owns which morphology representation/descriptors are perceptually meaningful; #150 owns the geometry/learnability question once a representation is independently justified.

For any retained `V(c)`, #150 will estimate

```text
P_J = D V^T D V
```

and test whether it is well scaled by `G` on resolved patches. This also formalizes a limitation already implicit here: because `c` has only two real DOF, the descriptor image is generally a 2D surface inside the larger morphology-feature space. Exact arbitrary feature targets may be unreachable.

Please treat `P_J` as a **local sensitivity diagnostic**, not as a reason to redesign the descriptors until they match `G`. Reflection-invariant morphology descriptors are expected to have null directions on symmetry loci, so a global positive lower bound is neither required nor possible for every descriptor family.

**Comment by rhythmatician** [comment]
## Reachability refinement from #150/#151

The existing warning that four morphology axes are not four independently controllable `c` coordinates can be made local and quantitative.

For fixed Julia presentation,

```text
V : R^2 -> R^d
rank(DV) <= 2.
```

So no matter how many retained perceptual descriptors `V` contains, the instantaneous first-order appearance tangent space has dimension at most two.

Use #151's local sensitivity probe/SVD to estimate the reachable tangent plane. If

```text
DV G^(-1/2) = U Sigma W^T,
```

then the retained non-null columns `U_r` span the local first-order appearance directions. For a candidate appearance error

```text
e = y_target - V(c),
```

record

```text
e_parallel = U_r U_r^T e
e_perp     = (I - U_r U_r^T) e.
```

Interpretation:

- `e_parallel` is locally actionable to first order;
- `e_perp` says the current tangent plane cannot address that component immediately;
- **do not** automatically delete `e_perp` from a later objective or declare it globally unreachable, because the appearance manifold can curve and the tangent plane changes along trajectories.

This suggests a useful separation for later learning:

```text
slow region selection: choose neighborhoods whose reachable appearance manifold fits the desired morphology
fast/local control: move through the one/two available tangent directions
```

Keep the perceptual descriptor space rich enough to test H1-H4; compress the **control interpretation**, not necessarily the evaluator.

Add #151 measurements before assuming an arbitrary descriptor vector is a valid fast-control target.
